core/domain/rights_manager.py

```diff
@@ -1036,6 +1183,8 @@ def _change_activity_status(
         commit_message: str. The human-written commit message for this change.
     """
     activity_rights = _get_activity_rights(activity_type, activity_id)
+    # Ruling out the possibility of None for mypy type checking.
+    assert activity_rights is not None
```

Comment:
Same question.

Reply:
Similarly, we are assessing attributes without checking for none cases.

Reply:
Ditto.

Reply:
This `_change_activity_status` method is called in `_publish_activity` and `_unpublish_activity` methods, and there we are already checking the none case of activity_rights by calling https://github.com/oppia/oppia/blob/0e69c2bcd7a91ef3595da274d060608efd3140ae/core/domain/rights_manager.py#L1080
https://github.com/oppia/oppia/blob/0e69c2bcd7a91ef3595da274d060608efd3140ae/core/domain/rights_manager.py#L1112

Reply:
Similar to above, this is a better justification than before, but better if you do an `if activity_rights is None` here (or use the "strict" paradigm where an exception is thrown in _get_activity_rights if the item doesn't exist).

Reply:
"strict" paradigm is added to _get_activity_rights and strict=True used here, Now _get_activity_rights will throw it's own error ( if activity_rights is not found for the given activity_id ).

Reply:
Sounds good, but please also update the docstring to cover the new Exception.

Reply:
Done!